src/test/java/org/sagebionetworks/web/client/security/AuthenticationControllerImplTest.java

```diff
@@ -283,6 +283,7 @@ public void testLoginUserNotAcceptedTermsOfUse() {
 
     assertNull(authenticationController.getCurrentUserProfile());
     verify(mockSessionDetector).initializeAccessTokenState();
+    verify(mockQueryClient).clear();
```

Comment:
This test represents the specific scenario captured in the Jira issue, so this specific verification should prove the bug is fixed.